aten/src/ATen/native/Copy.cpp

```diff
@@ -37,6 +38,19 @@ bool copy_transpose_valid(const at::Tensor& self, const at::Tensor& src) {
 namespace at {
 namespace native {
 
+Tensor & copy_(Tensor & self, const Tensor & src, bool non_blocking) {
+  Tensor b_src;
+  if (self.is_sparse() && src.is_sparse()) {
+    return at::copy_sparse_to_sparse_(self, src, non_blocking);
+  }
+  if (!self.is_sparse() && !src.is_sparse()) {
+    std::tie(b_src) = expand_inplace(self, src, "copy");
+    return s_copy_(self, b_src, non_blocking);
+  }
+  AT_ERROR("copy_() between dense and sparse Tensors is not implemented! Found self type = ",
```

Comment:
I know this code is just moved, but this isn't accurate anymore -- we have MKLDNN tensors, XLA tensors, etc.  How are these supposed to be implemented?

Reply:
That's a good point, I'll follow up on it.